src/utilitis/detectionsConverter.ts

```diff
@@ -0,0 +1,82 @@
+import { DetectionConvertion } from './../interfaces/IDetectorOpts';
+import Logger from './logger';
+
+const _filterDetections = (objectFilter: Array<String>, detections: Array<any>): Array<any> =>
+  objectFilter.length === 0
+    ? detections
+    : detections.filter(({ label }) => objectFilter.some(x => x === label));
+
+interface ImageSize {
+  width: number;
+  height: number;
+}
+
+interface ConverterOpts {
+  preview_image_size: ImageSize;
+  convertDetections?: DetectionConvertion;
+  objectFilter?: Array<String>;
+  frame?: any;
+}
+
+export default class DetectionsConverter {
+  _detections: Array<any>;
+  _framingBBox: { x: number; y: number; height: number; width: number };
+  _detectionConvertion: DetectionConvertion;
+  _preview_image_size: ImageSize;
+  _objectFilter: Array<String> = ['head', 'person'];
+
+  constructor(detections: Array<any>, converterOpts: ConverterOpts) {
+    this._detections = detections;
+    this._preview_image_size = converterOpts.preview_image_size;
+    if (converterOpts.objectFilter) {
+      this._objectFilter = converterOpts.objectFilter;
+    }
+    this._framingBBox = (converterOpts.frame && converterOpts.frame.bbox) || undefined;
+    this._setDetectionConvertion(converterOpts.convertDetections);
+  }
+
+  _setDetectionConvertion(detectionConvertion: DetectionConvertion) {
+    if (detectionConvertion === DetectionConvertion.FRAMING && !this._framingBBox) {
+      Logger.warn(
```

Comment:
Is this warning redundant?

Reply:
Yes I think this might be something that was there for old SW.